src/subdomains/supporting/bank/bank-account/is-dfx-iban.validator.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@nestjs/common';
+import {
+  registerDecorator,
+  ValidationArguments,
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+} from 'class-validator';
+import ibantools from 'ibantools';
+
+export const REQUEST_CONTEXT = '_requestContext';
+
+@ValidatorConstraint({ name: 'IsDfxIban', async: true })
+@Injectable()
+export class IsDfxIbanValidator implements ValidatorConstraintInterface {
+  private blockedIbans = ['LT..37800000', 'AT..14200200', 'AT..20602099', 'LT..60378000'];
+
+  async validate(iban: string) {
+    try {
+      iban = iban.split(' ').join('');
+
+      //Iban Tools
+      const { valid } = ibantools.validateIBAN(iban);
+      if (!valid) return false;
+
+      // check blocked IBANs
+      const blockedIban = this.blockedIbans.find((i) => i.split(' ').join('').toLowerCase() === iban);
+      if (blockedIban) return false;
+    } catch (e) {
+      return false;
+    }
+
+    return true;
+  }
+
+  defaultMessage() {
+    return `IBAN not valid`;
+  }
+}
```

Comment:
If you want, you can return different messages for different errors. Add a validation function, which will return a string in case of error and undefined otherwise. Then call this method from `validate` and `defaultMessage`(use `ValidationArguments`).